### ISSUE TITLE
Traefik does not refresh backend list on state change of consul catalog health checks

### ISSUE THREAD

**Issue by thebinary** [containous/traefik#3380]
### Do you want to request a *feature* or report a *bug*?

*bug*

### What did you do?
I have a uwsgi service that is registered using a local consul agent with local health checks for the uwsgi service. When I stop the uwsgi service on a node, consul detects the node to be unhealthy for that service but traefik does not remove it from the backend list.

Taking the uwsgi service offline
- stop the uwsgi service
- consul detects unhealthy node
- traefik does not remove from the backend list

Taking the uwsgi service online
- start the uwsgi service
- consul detects the healthy node
- traefik does not add the node on the backend list

### What did you expect to see?
Traefik should have added and removed nodes in the backend list according to consul catalog health check state changes.

### What did you see instead?
Traefik keeps the backend list as it is during the last restart/reload of traefik service.

### Output of `traefik version`: (_What version of Traefik are you using?_)

Traefik version
```
Version:      v1.6.1
Codename:     tetedemoine
Go version:   go1.10.2
Built:        2018-05-14_07:16:56PM
OS/Arch:      linux/amd64
```

Consul version
```
Consul v1.1.0
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[consul]
endpoint = "sd1.mydomain:8500"
watch = true
prefix = "traefik"

[consulCatalog]
endpoint = "sd1.mydomain:8500"
exposedByDefault = true
domain = "mydomain"
prefix = "traefik"
watch = true
trace = true

[web]
address = ":8080"
readOnly = false

[accesslog]
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

No logs are seen because no change is detected on health checks state change.

**Comment by thebinary** [comment]
The "watchHealthState" function in the consulcatalog provider does detect the changes in health checks of consul catalog services. But traefik only reflects change in its configuration if a service is removed or added in the catalog i.e "DiscoveredServices", "MissingServices".

The function only detects healthy and unhealthy services, but not healthy/unhealthy node. Is this intended?

**Comment by mmatur** [comment]
After some discussion with @thebinary on issue #3157. I am agree that there is two issues.

**Comment by traefiker** [comment]
Closed by #3390.
